language-plutus-core/src/Language/PlutusCore/Constant/Make.hs

```diff
@@ -47,20 +53,34 @@ makeBuiltinBS size bs = checkBoundsBS size bs ? BuiltinBS () size bs
 makeBuiltinSize :: Size -> Size -> Maybe (Constant ())
 makeBuiltinSize size size' = checkBoundsSize size size' ? BuiltinSize () size
 
--- | Coerce a 'Bool' to the corresponding PLC's @boolean@.
-makeDupBuiltinBool :: Bool -> Value TyName Name ()
-makeDupBuiltinBool b = runQuote $ if b then getBuiltinTrue else getBuiltinFalse
+-- | Convert a 'Bool' to the corresponding PLC's @boolean@.
+makeBuiltinBool :: Bool -> Quote (Value TyName Name ())
+makeBuiltinBool b = if b then getBuiltinTrue else getBuiltinFalse
+
+-- | Convert a 'Bool' to the corresponding PLC's @boolean@.
+-- Does not preserve the global uniqueness condition.
+dupMakeBuiltinBool :: Bool -> Value TyName Name ()
```

Comment:
Agreed.

Reply:
I'd like to keep

```
unsafeDupMakeConstant :: TypedBuiltinValue Size a -> Value TyName Name ()
unsafeDupMakeConstant tbv = fromMaybe err $ dupMakeConstant tbv where
    err = error $ "unsafeDupMakeConstant: out of bounds: " ++ prettyString tbv
```

anyway as it's used several times as well.

Reply:
I'd find it cleaner if all the use sites used explicit `runQuote`. Alternatively, if it's always safe to create constants because they don't use names, then just doing `runQuote` internally and exposing the non-monadic results seems okay. It just feels weird exporting both versions.

Reply:
> Alternatively, if it's always safe to create constants because they don't use names

Whoops, I should rename that to `makeBuiltin`, because it constructs not only constants, but also `boolean`s. That's why both version are exported.

Reply:
`boolean`s do have names, though - which we should be careful about. How annoying is it to just pretend that even the constants have to be in `Quote`? It would just make things more consistent.

Reply:
In the latest commit (9b74de0) I removed `dupMakeConstant` and defined `makeBuiltin :: TypedBuiltinValue Size a -> Quote (Maybe (Value TyName Name ()))` instead. But I preserved `unsafeMakeBuiltin :: TypedBuiltinValue Size a -> Value TyName Name ()`, because it's used several times is the library, tests and executables.